dev/scripts/src/alluxio.org/build-distribution/cmd/generate-tarball.go

```diff
@@ -172,6 +172,12 @@ func buildModules(srcPath, name, moduleFlag, version string, modules map[string]
 	}
 }
 
+func buildCSI(srcPath string) {
+	chdir(filepath.Join(srcPath, "integration/docker/csi"))
+	run(fmt.Sprintf("compiling csi"), "go", "build")
```

Comment:
the environment that will build this tarball has go 1.12 installed, which matches my current local environment. when i try to run `go build` locally, i get an error saying a module requires go 1.17:
```
# golang.org/x/net/http2
../../../../pkg/mod/golang.org/x/net@v0.0.0-20210510120150-4163338589ed/http2/client_conn_pool.go:305:6: undefined: errors.Is
note: module requires Go 1.17
```

so we must upgrade the go version in the docker image that defines the build environment:
```
# golang for tooling
RUN wget https://dl.google.com/go/go1.12.linux-amd64.tar.gz && \
    tar -xvf go1.12.linux-amd64.tar.gz && \
    mv go /usr/local
```
in `dev/jenkins/Dockerfile-jdk8` and `dev/jenkins/Dockerfile-jdk11`

then apply the change by rebuilding the alluxio-maven docker image, then upgrade all the image versions to use the new one. i did this yesterday for a separate change to the dockerfiles: https://github.com/Alluxio/alluxio/pull/15154